optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/score/MeetingSchedulingConstraintProvider.java

```diff
@@ -57,8 +57,7 @@
     // ************************************************************************
 
     protected Constraint roomConflict(ConstraintFactory constraintFactory) {
-        return constraintFactory.forEachIncludingNullVars(MeetingAssignment.class)
-                .filter(leftAssignment -> leftAssignment.getRoom() != null)
+        return constraintFactory.forEach(MeetingAssignment.class)
```

Comment:
@Christopher-Chianelli is this anyhow related to the other changes in this PR?

Reply:
Nope, but the `forEachIncludingNullVars` did catch a bug because it didn't set cache/selection order/sorter manner (where it cause a NPE).  If you look at the join, you see this:

```

                .join(MeetingAssignment.class,
                        equal(MeetingAssignment::getRoom),
                        lessThan(MeetingAssignment::getId),
                        overlapping(assignment -> assignment.getStartingTimeGrain().getGrainIndex(),
                                assignment -> assignment.getStartingTimeGrain().getGrainIndex() +
                                        assignment.getMeeting().getDurationInGrains()))
```

this requires room and startingTimeGrain to both not be null. Thus the `forEachIncludingNullVars` + room not null filter should be replaced by `forEach`.